chalice/awsclient.py

```diff
@@ -90,22 +90,27 @@ def create_function(self,
             kwargs['Timeout'] = timeout
         if memory_size is not None:
             kwargs['MemorySize'] = memory_size
+        return self._create_or_update_function_with_retries(
+            'create_function', kwargs)['FunctionArn']
```

Comment:
* You can technically pass any client method here so the name seems too restrictive.
* We don't need getattr logic, it's always going to be a method we know statically.  I'd prefer to update this to just pass the method obj directly.  (`self._client('lambda').create_function`)